rclcpp/include/rclcpp/executor.hpp

```diff
@@ -371,6 +371,9 @@ class Executor
    *   If the time spent inside the blocking loop exceeds this timeout, return a TIMEOUT return
    *   code.
    * \return The return code, one of `SUCCESS`, `INTERRUPTED`, or `TIMEOUT`.
+   * \note This method will check the future and the timeout only when the executor is woke up.
+   * If this future is unrelated to an executor's entity, this method will not detect correctly
+   * when it's completed.
```

Comment:
Minor fix + indentation to match the rest:

```suggestion
   * \note This method will check the future and the timeout only when the executor is woken up.
   *   If this future is unrelated to an executor's entity, this method will not detect correctly
   *   when it's completed.
```

Should we also be more explicit about what could happen, e.g., "will not correctly detect when it's completed and therefore may wait forever and never time out."

Reply:
Done!